documents/model.go

```diff
@@ -851,20 +862,34 @@ func (m *CoreDocumentModel) IsAccountInRole(roleKey []byte, account identity.DID
 }
 
 // assembleTokenMessage assembles a token message
-func assembleTokenMessage(tokenIdentifier []byte, granterID []byte, granteeID []byte, roleID []byte, docID []byte) []byte {
+func assembleTokenMessage(tokenIdentifier []byte, granterID []byte, granteeID []byte, roleID []byte, docID []byte) ([]byte, error) {
+	tokenIdentifiers := [][]byte{tokenIdentifier, roleID, docID}
+	for _, id := range tokenIdentifiers {
+		if len(id) != IDByteCount {
+			return nil, errors.New("invalid identifier length")
+		}
+	}
```

Comment:
You should also check granter and grantee for the right length :)

Reply:
or maybe even better, use `identity.DID` type for it?

Reply:
that definitely makes sense

Reply:
yep, now that identity is merged in the bytelength is correct (20)

Reply:
👍